scripts/init.groovy.d/ab_configure_github_oauth.groovy

```diff
@@ -0,0 +1,115 @@
+// output init hook stage
+println '### JENKINS AUTOCONFIG: GITHUB OAUTH ###'
+
+/**
+  * Check if Jenkins should not configure itself and skip.
+  */
+
+Boolean skipConfiguringJenkins() {
+    !(System.env?.ENVIRONMENT in ['dev', 'staging']) ||
+        (new File("${Jenkins.instance.rootDir}/autoConfigComplete").exists())
+}
+if(skipConfiguringJenkins()) {
+    println 'Skipping Jenkins auto-configuration'
+    return
+}
+
+
+/**
+  * From this point onward Jenkins should configure itself.
+  */
+import jenkins.*
+import jenkins.model.*
+import hudson.*
+import hudson.model.*
+
+import org.yaml.snakeyaml.LoaderOptions
+import org.yaml.snakeyaml.Yaml
+import org.yaml.snakeyaml.constructor.SafeConstructor
+import software.amazon.awssdk.regions.Region
+import software.amazon.awssdk.services.secretsmanager.SecretsManagerClient
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueRequest
+import software.amazon.awssdk.services.secretsmanager.model.GetSecretValueResponse
+
+import hudson.security.SecurityRealm
+import org.jenkinsci.plugins.GithubSecurityRealm
+import net.sf.json.JSONObject
+
+Map getSecret(Map options) {
+    String secretName = options.secretId
+    Region region = Region.of(options.region)
+    SecretsManagerClient client = SecretsManagerClient.builder()
+        .region(region)
+        .build()
+    GetSecretValueRequest getSecretValueRequest = GetSecretValueRequest.builder()
+        .secretId(secretName)
+        .build()
+    GetSecretValueResponse getSecretValueResponse = client.getSecretValue(getSecretValueRequest)
+    String secret = getSecretValueResponse.secretString()
+    LoaderOptions loaderOptions = new LoaderOptions()
+    def yaml = new Yaml(new SafeConstructor(loaderOptions))
+    yaml.load(secret)
+}
+String ENVIRONMENT = System.env?.ENVIRONMENT
+String HALF = System.env?.HALF
+
+// list of github realm maps
+github_realm = getSecret(region: 'us-east-1', secretId: "jenkins-ng/oauth/github")."${ENVIRONMENT}"."${HALF}".collectEntries()
```

Comment:
Question: why is `github_realm` part of the Groovy binding and not strongly typed to `Map`?

Reply:
it will get overwritten here if its not a binding:
https://github.com/integralads/jenkins-bootstrap-shared/pull/4/files/9c3329cc0f8eef912b2a203fa57a79edb2e51800#diff-a75fd369c4490f90becf868da95bd857c2b02197eca3c5a56d9af5413fbb3b06R82-R84